pkg/monitoring/metrics.go

```diff
@@ -40,6 +46,36 @@ func SetReadyGauge(isReady bool) {
 	}
 }
 
+func setKubemacpoolDeployedGauge(isDeployed bool) {
+	if isDeployed {
+		kmpDeployedGauge.Set(1)
+	} else {
+		kmpDeployedGauge.Set(0)
+	}
+}
+
+func TrackMonitoringComponents(conf *cnao.NetworkAddonsConfigSpec, monitoringAvailable bool) error {
+	if !monitoringAvailable {
+		return nil
+	}
+
+	err := trackKubemacpoolMonitoring(conf)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func trackKubemacpoolMonitoring(conf *cnao.NetworkAddonsConfigSpec) error {
+	isKubemacpoolDeployed := false
+	if conf.KubeMacPool != nil {
+		isKubemacpoolDeployed = true
+	}
+	setKubemacpoolDeployedGauge(isKubemacpoolDeployed)
+	return nil
```

Comment:
not needed

Reply:
not sure what you mean here.. the function will return []error in general just so we have the option to return errors in the future.

Reply:
i don't think we need to create infra for something that is not used right now, its a simple change that if someone need he can add a return value
atm, there is no other returns beside this one

Reply:
I disagree. Returning `[]error/error` from functions is generally a good practice. why not here?

Reply:
The function has only one return statement and its always return nil
there is no good practice at all of returning the same thing no matter what the function is doing
i bet there are even static analyse tools that will warn / fail / optimize it

imo there should not be redundant code

Reply:
I tend to prefer adding the infra beforehand.
`TrackMonitoringComponents()` is the house for handling monitoring for all deployed components.
Each can potentially return an error, so `TrackMonitoringComponents()` collects the errors from them and flattens them to one error line if needed. 
Granted, currently there is only one: `trackKubemacpoolMonitoring()`, but I prefer that we lay out how we want to function to look like from the beginning, so that next time we want to add another monitoring (for knmstate for example) then that developer would only need to add the component's function.

We did the same when we added the `SpecialCleanUp()` [here](https://github.com/kubevirt/cluster-network-addons-operator/pull/270), and I think it's what we should do here as well.